analyses/tumor-purity-exploration/util/function_filter-by-threshold.R

```diff
@@ -0,0 +1,39 @@
+
+
+
+#' Filter an expression data frame or matrix by median cancer group tumor purity
+#'
+#' @param expression_data A data frame or matrix containing expression data. Columns are expected to be genes, 
+#'   and rows are expected to be samples. 
+#'
+#' @return An updated `expression_data` object that contains expression data only for
+#'  samples that pass the threshold
+filter_expression_by_tumor_purity <- function(expression_data) {
+  
+  # tumor purity metadata file which has been filtered to only biospecimens that
+  #  survive the cancer-group-level threshold
+  tumor_purity_file <- file.path(
+    rprojroot::find_root(rprojroot::has_dir(".git")),
+    "analyses",
+    "tumor-purity-exploration",
+    "results",
+    "thresholded_rna_stranded_same-extraction.tsv"
```

Comment:
I think this is _fine_ for this particular function, but registering my discomfort with the practice of hardcoding this file in here instead of using an argument 🙃

Reply:
Don't disagree with you! This could also be an argument with that path as default.

Reply:
Noting that I did that ^

Reply:
^ Noting I disagreed with myself since I decided I don't like having such a prescribed path, even if relative, as an argument. I did more things in https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1663/commits/c96f96d3747cf886becb1b49f20ace75d4b3d691 and named argument to be named `bs_ids_file` and not have default. Instead, the file w/ ids to filter to can be specified in scripts that use it, which need to spell out a lot of the relevant to source this function anyways!